harness/src/harness/build_agent.py

```
@@ -0,0 +1,285 @@
+        self._timeout = timeout
+
+    def _argv(self, prompt: str) -> list[str]:
+        if self._provider == "codex":
+            return ["codex", "exec", prompt]
+        return ["claude", "-p", prompt]
+
+    def compile(self, message: str, tool_id: str | None = None) -> WorkflowSpec:
+        prompt = _CLI_SCHEMA_PROMPT + "\n\nOPERATOR REQUEST:\n" + message.strip()
+        if tool_id:
+            prompt += f"\n\n(Refine the existing tool: {tool_id})"
+        proc = subprocess.run(self._argv(prompt), capture_output=True, text=True, timeout=self._timeout)  # noqa: S603
```

Ruff (0.15.18)

[warning] `subprocess.run` without explicit `check` argument

Add explicit `check=False`

(PLW1510)